lib/sequent/core/workflow.rb

```diff
@@ -5,7 +5,31 @@ module Core
     class Workflow
       include Helpers::MessageHandler
 
+      def self.on(*message_classes, &block)
+        decorated_block = ->(event) do
+          begin
+            old_event = Thread.current[:handling_event]
+            Thread.current[:handling_event] = event
+            self.instance_exec(event, &block)
+          ensure
+            Thread.current[:handling_event] = old_event
+          end
+        end
+        super(*message_classes, &decorated_block)
+      end
+
+      def current_event
+        Thread.current[:handling_event]
+      end
+
       def execute_commands(*commands)
+        commands.each do |command|
+          if command.respond_to?(:event_aggregate_id) && command.event_aggregate_id.blank?
```

Comment:
Can we move this to the `Sequent.configuration.command_service.execute_commands`? This way it does not matter if the shortcut `execute_commands` or the real `execute_commands` is called.